File: src/Python/LC5.py
class Solution:
    def isPalindrome(self, s, l):
        for i in range(l//2):
            if s[i] != s[l - 1 - i]:
                return False
        return True

    def longestPalindrome(self, s):
        if len(set(s)) == 1:
            return s
        length = len(s)
        result = ""
        for i in range(length):
            for j in range(i, length):
                substring = s[i:j + 1]
                l = len(substring)
                if self.isPalindrome(substring, l) and l > len(result):
                    result = substring
        return result

print(Solution().longestPalindrome("hello"))
print(Solution().longestPalindrome("firefightahtenetdestroy"))
print(Solution().longestPalindrome("racecar-redrum-nitin"))

File: src/C++/LC5.cpp
#include <iostream>
#include <conio.h>
#include <string_view>

class Solution
{
    public:
    bool isPalindrome(std::string_view s, int l)
    {
        for(int i = 0; i < l/2; i++)
        {
            if(s[i] != s[l - 1 - i])
                return false;
        }
        return true;
    }

    std::string longestPalindrome(std::string_view s)
    {
        std::string result = "";
        std::string substring;
        int length = s.size();
        for(int i = 0; i < length; i++)
            for(int j = i; j < length; j++)
            {
                substring = s.substr(i, j - i + 1);
                int l = substring.size();
                if(isPalindrome(substring, l) && l > result.size())
                    result = substring;
            }
        return result;
    }
};

int main()
{
    std::cout<<Solution().longestPalindrome("hello")<<"\n";
    std::cout<<Solution().longestPalindrome("firefightahtenetdestroy")<<"\n";
    std::cout<<Solution().longestPalindrome("racecar-redrum-nitin");
    getch();
    return 0;
}